static/js/services/count-unread-records.js

```diff
@@ -0,0 +1,36 @@
+var _ = require('underscore'),
+    TYPES = [ 'report', 'message' ];
+
+angular.module('inboxServices').factory('CountUnreadRecords', function(
+  $q,
+  DB
+) {
+
+  'use strict';
+  'ngInject';
+
+  var getTotal = function() {
+    return DB().query('medic-client/data_records_by_type', { group: true });
+  };
+
+  var getRead = function() {
+    return DB({ meta: true }).query('medic-user/read', { group: true });
+  };
+
+  var getValue = function(response, type) {
+    var result = _.findWhere(response.rows, { key: type });
+    return (result && result.value) || 0;
+  };
+
+  return function() {
+    return $q.all([ getTotal(), getRead() ]).then(function(results) {
+      var total = results[0];
+      var read = results[1];
+      var result = {};
+      TYPES.forEach(function(type) {
+        result[type] = getValue(total, type) - getValue(read, type);
```

Comment:
This section is confusing to me, but I suspect it's my lack of familiarity with how CouchDB handles reduce. Once I played around on the command line I understood it better. I don't know if that means that's good enough, or if this could be reworked to be more generally understandable.